python_modules/libraries/dagster-sling/dagster_sling/components/sling_replication_collection/component.py

```diff
@@ -133,11 +118,25 @@ def build_asset(
     ) -> AssetsDefinition:
         op_spec = replication_spec_model.op or OpSpec()
 
+        translator = replication_spec_model.translator or DagsterSlingTranslator()
+
+        class ReplicationTranslatorWithCodeReferences(DagsterSlingTranslator):
+            def get_asset_spec(self, obj: Any) -> AssetSpec:
+                asset_spec = translator.get_asset_spec(obj)
+                return merge_code_references(
+                    asset_spec,
+                    [
+                        LocalFileCodeReference(
+                            file_path=str(context.path / replication_spec_model.path),
+                        )
+                    ],
+                )
```

Comment:
i think this would be a lot less gross if we moved to `translation` fn approach we do elsewhere, should we stack this change on that or just do it later?

Reply:
yea good call https://github.com/dagster-io/dagster/pull/29878

Reply:
Hmm, I still end up having to construct a wrapper translator in `build_defs` in order to pass along the `ComponentLoadContext.path`, which isn't available in the resolution context etc.

Reply:
what if you changed `def translator` from a `@property` to `def get_translator` that takes the info you need and instantiates `ProxyDagsterSlingTranslator` with that and optional `translation` ? 

Feel like its worth avoiding the in line class calling other class thing if something like that feels clean but this isn't a hard block